### PR TITLE
go to v1.0.34 of openshift-pipeline

### PR DESCRIPTION
**Pull request by gabemontero**
pulls in the openshift-pipeline fix for https://bugzilla.redhat.com/show_bug.cgi?id=1395555 for jenkins-*-centos7

@bparees FYI

@tdawson we'll need a new RPM for v1.0.34 of openshift-pipeline (the other RPMs are up to date) and a respin of jenkins-*-rhel7 on brew-pulp - thanks

@dongboyan77 FYI

### CODE REVIEW

**Comment by gabemontero** [comment]
[test]

**Comment by gabemontero** [comment]
on these intermittent test job failures, we are not dumping the container logs cause of that rc == 143 check after the Testing HTTP Connection... does not get the expected results ... I'm leaning toward adding a change to our test script that just always dumps the jenkins container.

**Comment by bparees** [comment]
I really don't want to do that, it makes the runs incredibly messy.  Let's make it conditional.  set the conditional to false at the end of the test script (before we print SUCCESS at the very end) so we don't dump the container if all the tests were successful.

we should also get rid of the echo "  Success!" within the run_tests function, because it's weird to print success in multiple places (one for each test, one overall).

if we want to print something for each test, let's make the two messages obvious that one is test case success and one is overall success.

**Comment by tdawson** [comment]
rpm built:
  jenkins-plugin-openshift-pipeline-1.0.34-1.el7
images built:
  openshift3/jenkins-1-rhel7:1.651.2-34
  openshift3/jenkins-2-rhel7:2.7-19

**Comment by gabemontero** [comment]
thanks @tdawson !

**Comment by bparees** [comment]
[test]

**Comment by gabemontero** [comment]
running some local tests with the conditional dumping of container logs as prescribed by @bparees

**Comment by gabemontero** [comment]
Interesting ... I saw an issue with "Testing HTTP Connection..." locally when testing the conditional container dump.  Not sure what to make of it, but it would be curious if the same thing is happening with the PR testing:

```
INFO: Started all plugins
Nov 17, 2016 12:57:12 AM hudson.ExtensionFinder$GuiceFinder$FaultTolerantScope$1 error
WARNING: Failed to instantiate optional component org.jenkinsci.plugins.workflow.steps.scm.GitStep$DescriptorImpl; skipping
Nov 17, 2016 12:57:12 AM io.fabric8.kubernetes.client.Config tryServiceAccount
WARNING: Error reading service account token from: [/var/run/secrets/kubernetes.io/serviceaccount/token]. Ignoring.
```

**Comment by gabemontero** [comment]
rhel mirror repo issues now ... trying one more time ... [test]

**Comment by gabemontero** [comment]
still with the mirrors ... will try again either later tonight or tomorrow AM

**Comment by tdawson** [comment]
There was a problem when syncing to the mirrors.  It's been fixed and things should work now.

**Comment by gabemontero** [comment]
thanks for the info @tdawson ... [test]

File: 1/test/run [hunk 1]
```diff
@@ -11,6 +11,8 @@ shopt -s nullglob
 
 IMAGE_NAME=${IMAGE_NAME-openshift/jenkins-1-centos7-candidate}
 
+FAILURE=`echo true`
```
Comment (gabemontero) [inline 1/test/run]: It was causing funny highlighting in my emacs bash editor and i wanted to make sure i got something functional to test ... I'll work on switching

**Comment by bparees** [comment]
it looks to me like we're failing because jenkins doesn't quite start in 1 minute (perhaps due to our heap changes, new plugins, etc):

```
Testing HTTP connection...
Sending GET request to http://172.17.0.2:8080/
Sending GET request to http://172.17.0.2:8080/
Sending GET request to http://172.17.0.2:8080/
Sending GET request to http://172.17.0.2:8080/
Sending GET request to http://172.17.0.2:8080/
Sending GET request to http://172.17.0.2:8080/
Sending GET request to http://172.17.0.2:8080/
Sending GET request to http://172.17.0.2:8080/
Sending GET request to http://172.17.0.2:8080/
Sending GET request to http://172.17.0.2:8080/
Stopping and removing container 8f323239f946244d9193d136dcb73d61ded0c871dc43b858129271a7370bd858...
8f323239f946244d9193d136dcb73d61ded0c871dc43b858129271a7370bd858
Dumping logs for 8f323239f946244d9193d136dcb73d61ded0c871dc43b858129271a7370bd858
```

I suggest we bump the number of retries to 12 (it waits 10 seconds between each retry)

**Comment by gabemontero** [comment]
The new form of dumping the container on an error appears to have been informative.  Best as I can tell, the http test connection simply gave up too soon.  There were two instances.  In the second, it seemed obvious that Jenkins was in the middle of its startup.  The first, it seemed to come up, but can't say for sure if it finished coming up between when the test stopped and when the container was dumped.

I'm going to try and increase the time period, just to see what effect it has (assuming the consistent failures this evening continue)

**Comment by bparees** [comment]
heh, sounds like we're on the same page :)

**Comment by gabemontero** [comment]
ok ... our common analysis passed through the ether :-)

**Comment by bparees** [comment]
one other strange thing is it appears the output is comingled, not sure why that is but it makes it a lot harder to follow what failed and what logs are from what test.

**Comment by bparees** [comment]
either that or we're actually getting the wrong container id, which would explain the failures better.

see:
```
Created container e763f2e93b2962138c7fefd0a0a523611e341e0a89d5883294270afbd3439e7d
Testing HTTP connection...
Sending GET request to http://172.17.0.2:8080/
Sending GET request to http://172.17.0.2:8080/
Sending GET request to http://172.17.0.2:8080/
Sending GET request to http://172.17.0.2:8080/
Sending GET request to http://172.17.0.2:8080/
Sending GET request to http://172.17.0.2:8080/
Sending GET request to http://172.17.0.2:8080/
Sending GET request to http://172.17.0.2:8080/
Sending GET request to http://172.17.0.2:8080/
Sending GET request to http://172.17.0.2:8080/
Stopping and removing container 8f323239f946244d9193d136dcb73d61ded0c871dc43b858129271a7370bd858...
8f323239f946244d9193d136dcb73d61ded0c871dc43b858129271a7370bd858
```

**Comment by gabemontero** [comment]
looks like cleanup() dumps *every* container that was created

**Comment by bparees** [comment]
hm. good catch.

**Comment by gabemontero** [comment]
we *could* dump the container in test_connection if $result is non-zero ... be more precise on which container we had trouble with

**Comment by bparees** [comment]
at a minimum we should be printing a message in there indicating things failed(ie it's giving up after 10 retries), because right now it takes too much detective work to realize that what failed was it hitting its limit of 10 retries and bombing out.

trying to dump the specific container might be too much work (given all the ways/places things could fail.  it'd be easy to solve for this one, but not generally for all the tests)

**Comment by gabemontero** [comment]
yes agreed easy for this case (since the container name is passed into test_connection by its callers, even if it is not used currently) but in general more complicated

btw, it passed with the longer grace period

merge now to get the centos images updated, and continue tweaking the debug in a separate pull ?

**Comment by bparees** [comment]
yeah. (merged)

**Pull request merged**